### PR TITLE
[feat] Support Google Analytics 4 properties

### LINKED ISSUE #1332: [Feature]: Support Google Analytics 4 properties
### Requirement

Starting July 1, 2023 Google will be transitioning to GA 4 properties.
The exact message on GA home page: 
```
On July 1, 2023, this property will stop processing data. Starting in March 2023, for continued website measurement, migrate your original property settings to a Google Analytics 4 (GA4) property, or they'll be copied for you to an existing GA4 property, reusing existing site tags.
```

### Problem

I created a GA4 version of my original Jaeger UI property and tried to use it but no luck. Looks like the react-ga package that is used in this repo does not support GA tags (Vs UA ones).
Found an issue in that repo: https://github.com/react-ga/react-ga/issues/541

### Proposal

I have not researched it enough, but looks like react-ga4 package might work for GA tags.
https://github.com/codler/react-ga4

### Open questions

_No response_

### PR DESCRIPTION
**Pull request by anikdhabal**
## Which problem is this PR solving?
 Resolves #1332 

## Short description of the changes
- 
Support Google Analytics 4 properties

### CODE REVIEW

**Comment by anikdhabal** [comment]
@yurishkuro can you review this PR?

File: package.json [hunk 1]
```diff
@@ -63,5 +63,8 @@
     "hooks": {
       "pre-commit": "npm-run-all -ln --parallel lint test"
     }
+  },
+  "dependencies": {
+    "react-ga4": "^2.1.0"
```
Comment (yurishkuro) [inline package.json:68]: is it needed here? already added at the lower-level package.json

File: packages/jaeger-ui/src/utils/tracking/ga.tsx [hunk 2]
```diff
@@ -56,7 +56,7 @@ const GA: IWebAnalyticsFunc = (config: Config, versionShort: string, versionLong
       msg = `jaeger/${msg}`;
     }
     msg = msg.slice(0, 149);
-    ReactGA.exception({ description: msg, fatal: false });
```
Comment (yurishkuro) [inline packages/jaeger-ui/src/utils/tracking/ga.tsx:59]: does the new module not support a distinction between exceptions and other events?  The `send` in L155 specifies `hitType`, should this one also have `hitType=exception`?

**Review by yurishkuro — DISMISSED** [review]
please remove `package-lock.json` from PR

File: packages/jaeger-ui/src/utils/tracking/ga.tsx (hunk 2 above)
Comment (anikdhabal) [reply packages/jaeger-ui/src/utils/tracking/ga.tsx:59]: Ahh My bad this should be hitType=exception . Thank you.

File: package.json (hunk 1 above)
Comment (anikdhabal) [reply package.json:68]: ok ok

**Review by yurishkuro — APPROVED** [review]
lgtm. Did you test it with a real GA ID?

**Comment by anikdhabal** [comment]
> lgtm. Did you test it with a real GA ID?

Yes, it is fine.

**Comment by yurishkuro** [comment]
> Yes, it is fine.

can you include some screenshots?

**Comment by anikdhabal** [comment]
> > Yes, it is fine.
> 
> can you include some screenshots?

Okk I will share some result.

**Comment by yurishkuro** [comment]
@anikdhabal this PR does not include any configuration changes, so the same GA ID config will be used. Will it work if someone has the old GA ID in their config, which is not GA4?

**Comment by anikdhabal** [comment]
> @anikdhabal this PR does not include any configuration changes, so the same GA ID config will be used. Will it work if someone has the old GA ID in their config, which is not GA4?

No, it doesn't work for the old GA ID; it only works for GA4. I haven't found a straightforward way to support the old GA configuration. Let me do a bit more research about it or talk to the maintainer to include the configuration.

**Comment by yurishkuro** [comment]
> it only works for GA4. I haven't found a straightforward way to support the old GA configuration

Should we keep using both libraries and switch depending on the version of GA ID? Do you know what the timeline for Google to completely phase out old IDs?

**Comment by anshgoyalevil** [comment]
According to [Google](https://support.google.com/analytics/answer/10089681?hl=en&ref_topic=12154439,12153943,2986333,&sjid=17505237808164032051-AP&visit_id=638285326746816164-286965898&rd=1),

> Starting July 1, 2023, standard UA properties [will stop processing data](https://support.google.com/analytics/answer/11583528)

**Comment by yurishkuro** [comment]
Yeah, then I think it's fine to drop support for UA, and only support GA4

**Comment by anikdhabal** [comment]
> Yeah, then I think it's fine to drop support for UA, and only support GA4

Yeah it's fine.

**Comment by yurishkuro** [comment]
CI is still failing on this pr

**Comment by anikdhabal** [comment]
> CI is still failing on this pr

I am fixing it now.

**Comment by yurishkuro** [comment]
Fixed in #2071

**Pull request closed without merging**